src/AzureAuth.Test/ExceptionListToStringConverterTest.cs

```diff
@@ -0,0 +1,407 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.AzureAuth.Test
+{
+    using System;
+    using System.Collections.Generic;
+    using FluentAssertions;
+    using Microsoft.Identity.Client;
+    using NUnit.Framework;
+
+    public class ExceptionListToStringConverterTest
+    {
+        [Test]
+        public void EmptyExceptionList()
+        {
+            IEnumerable<Exception> exceptions = new List<Exception>();
+
+            // Act
+            string result = ExceptionListToStringConverter.Execute(exceptions);
+
+            // Assert
+            result.Should().BeNull();
+        }
+
+        [Test]
+        public void NullExceptionList()
+        {
+            IEnumerable<Exception> exceptions = null;
+
+            // Act
+            string result = ExceptionListToStringConverter.Execute(exceptions);
+
+            // Assert
+            result.Should().BeNull();

```

Comment:
I think this case should throw - if given a null list to begin with, that likely indicates a bug. I don't think our callers have a valid scenario where the list is actually null and I don't think we want that to be supported.